src/org/labkey/test/tests/component/EditableGridTest.java

```diff
@@ -434,6 +482,717 @@ public void testExpandedPasteIntoSkinnySelection() throws Exception
+                    .build()
+                    .perform();
+        }
+
+        actions = new Actions(getDriver());
+        actions.pause(500)
+                .sendKeys(Keys.ENTER)
+                .build()
+                .perform();
+
+        checker().verifyTrue("TextArea should have gone away after hitting <Enter>.",
+                shortWait().until(ExpectedConditions.stalenessOf(editCell)).booleanValue());
+
+        scrollIntoView(editableGrid.getCell(0, lastColumnName));
+
+        WebElement updatedGridCell03 = editableGrid.getCell(0, PASTE_ML);
+        checker().verifyTrue("Cell not updated with many short lines.",
+                waitFor(()->!updatedGridCell03.getText().isEmpty(), 1_000));
+
+        log("Single line updated cell Width & Height: " + updatedGridCell03.getSize());
+
+        checker().verifyTrue(String.format("Height of updated cell with many short lines, %d, is not as expected %d (+/- 5).",
+                        updatedGridCell03.getSize().height, maxHeight),
+                Math.abs(updatedGridCell03.getSize().height - maxHeight) <= 5);
+
+        checker().verifyTrue(String.format("Width of updated cell many short lines, %d, is not as expected %d (+/- 10).",
+                        updatedGridCell03.getSize().width, emptyWidth),
+                Math.abs(updatedGridCell03.getSize().width - emptyWidth) <= 10);
+
+        checker().screenShotIfNewError("Many_Short_Line_Size_Error");
+
+        log("Reset the grid for the last time.");
+        editableGrid.selectAll(true);
+        editableGrid.clickRemove();
+
+        log("Validate <esc> exits edit mode and does not save.");
+        editableGrid.addRows(1);
+        scrollIntoView(editableGrid.getCell(0, lastColumnName));
+
+        editCell = editableGrid.doubleClickMultiLineCell(0, PASTE_ML);
+
+        for(int rows = 0; rows < 5; rows++)
+        {
+            actions = new Actions(getDriver());
+            actions.sendKeys("z")
+                    .keyDown(Keys.SHIFT)
+                    .sendKeys(Keys.ENTER)
+                    .keyUp(Keys.SHIFT)
+                    .build()
+                    .perform();
+        }
+
+        actions = new Actions(getDriver());
+        actions.pause(500)
+                .sendKeys(Keys.ESCAPE)
+                .build()
+                .perform();
+
+        checker().verifyTrue("TextArea should have gone away after hitting <Esc>.",
+                shortWait().until(ExpectedConditions.stalenessOf(editCell)).booleanValue());
+
+        scrollIntoView(editableGrid.getCell(0, lastColumnName));
+
+        WebElement updatedGridCell04 = editableGrid.getCell(0, PASTE_ML);
+        checker().verifyTrue("Cell should not be updated after hitting <esc>.",
+                waitFor(()->updatedGridCell04.getText().isEmpty(), 1_000));
+
+        checker().screenShotIfNewError("Exit_With_Escape_Error");
+
+    }
+
+    static final Locator SELECTED_CELL_LOCATOR = Locator.tagWithClass("div", "cell-selection");
```

Comment:
Extract the common locator in `EditableGrid.getSelectionCellTexts` and expose a new method which returns the `List<WebElement>` for the selected cells.

Reply:
Was able to leverage functionality in the editable grid for this.
This has been removed.